commit e5c8bb0e76069d9afefcc9d96d8faca384cc67d2
Author: ddv2005 <ddv@abinet.com>
Date:   Mon Aug 20 10:48:38 2012 -0400

    Revert "Disable IRQs in dwc_otg_hcd_qh_free" because DWC_DMA_FREE should be called with enabled IRQs.

Index: linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c
===================================================================
--- linux-3.2.46.orig/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c	2013-07-26 19:35:17.000000000 +0000
+++ linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd_queue.c	2013-07-26 19:35:35.000000000 +0000
@@ -54,11 +54,10 @@
  */
 void dwc_otg_hcd_qh_free(dwc_otg_hcd_t * hcd, dwc_otg_qh_t * qh)
 {
-	dwc_irqflags_t flags;
 	dwc_otg_qtd_t *qtd, *qtd_tmp;
 
 	/* Free each QTD in the QTD list */
-	DWC_SPINLOCK_IRQSAVE(hcd->lock,&flags);
+	DWC_SPINLOCK(hcd->lock);
 	DWC_CIRCLEQ_FOREACH_SAFE(qtd, qtd_tmp, &qh->qtd_list, qtd_list_entry) {
 		DWC_CIRCLEQ_REMOVE(&qh->qtd_list, qtd, qtd_list_entry);
 		dwc_otg_hcd_qtd_free(qtd);
@@ -77,7 +76,7 @@
 	}
 
 	DWC_FREE(qh);
-	DWC_SPINUNLOCK_IRQRESTORE(hcd->lock,flags);
+	DWC_SPINUNLOCK(hcd->lock);
 	return;
 }
 
